src/Type/Constant/ConstantArrayType.php

```diff
@@ -80,6 +81,7 @@ public function __construct(
 		private array $valueTypes,
 		int|array $nextAutoIndexes = [0],
 		private array $optionalKeys = [],
+		private bool $isList = false,
```

Comment:
Does this need a new property? Can't we always figure it out from keyTypes and nextAutoIndexes?

Reply:
I tried that at first but it was a bit buggy. I've reworked some other parts so it might be doable now.

Reply:
The property is necessary, or at the very least I don't know how to resolve issues that arise otherwise:
- `getAllArrays()` should only return lists if the array is a list, which is unknowable otherwise
- other code sometimes relies on `ConstantArrayType` existing on its own and produces worse type inference with the accessory type separately